pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -1571,17 +1726,31 @@ func excludeTags(ctx context.Context, tags []model.TagFilters) []model.TagFilter
 		"error":               true,
 		"service.name":        true,
 	}
-	var newTags []model.TagFilters
+	newTags := make([]string, 0)
 	for _, tag := range tags {
-		_, ok := excludedTagsMap[tag.TagKeys]
+		_, ok := excludedTagsMap[tag]
 		if !ok {
 			newTags = append(newTags, tag)
 		}
 	}
 	return newTags
 }
 
-func (r *ClickHouseReader) GetTagValues(ctx context.Context, queryParams *model.TagFilterParams) (*[]model.TagValues, *model.ApiError) {
+func (r *ClickHouseReader) GetTagValues(ctx context.Context, queryParams *model.TagFilterParams) (*model.TagValues, *model.ApiError) {
+
+	if queryParams.TagKey.Type == model.TagTypeNumber {
+		return &model.TagValues{
+			NumberTagValues: make([]float64, 0),
+			StringTagValues: make([]string, 0),
+			BoolTagValues:   make([]bool, 0),
+		}, nil
+	} else if queryParams.TagKey.Type == model.TagTypeBool {
+		return &model.TagValues{
+			NumberTagValues: make([]float64, 0),
+			StringTagValues: make([]string, 0),
+			BoolTagValues:   []bool{true, false},
+		}, nil
+	}
```

Comment:
What's happening here?

Reply:
We are not suggesting numbers and boolean values can only be true and false so hard coded that to save a query.